src/python/pants/engine/internals/parser.py

```diff
@@ -498,6 +503,9 @@ def __init__(self, name: str) -> None:
         self.args: tuple[Any, ...] = ()
         self.kwargs: dict[str, Any] = {}
 
+    def __hash__(self) -> int:
+        return hash(self.name)
```

Comment:
Is this related to the parsing fix?

Reply:
Yes, the keys of a dictionary must be hashable, so in order to support this syntax:
```python
__defaults__({ resources: dict(...) })
```

The `resources` is an instance of `Registrar` so it must be hash-able.